src/proto/protocached.proto

```diff
@@ -0,0 +1,61 @@
+syntax = "proto3";
+
+package protocached;
+
+// Result for rpc response
+message ResponseResult {
+  enum ResultType {
+    SUCCESS = 0;
+    ERROR = 1;
+  }
+
+  // specify the response result
+  ResultType result = 1;
+  // optional error message if the result is ERROR
+  string errorMsg = 2;
```

Comment:
I thought the style guide standard is to use snake case for proto field name?

Also, shouldn't this be optional? If the result status is successful, there wouldn't be any error message, right?

Reply:
Yes, this is (and should be) optional. AFAIK, i think proto3 syntax default everything to be optional if you don't specify (at least within Google it's true), I will double check that.

Reply:
Yeah, in proto3 syntax, everything is optional (https://developers.google.com/protocol-buffers/docs/proto3#specifying-field-rules). They get rid of required/optional field. You can either be optional or repeated.

Also see: http://www.crankuptheamps.com/blog/posts/2017/10/12/protobuf-battle-of-the-syntaxes/